src/web/extractors.rs

```diff
@@ -584,6 +702,116 @@ mod tests {
         assert_eq!(result.full, true);
     }
 
+    #[test]
+    fn test_valid_bso_request() {
+        let payload = HawkPayload::test_default();
+        let state = make_state();
+        let header = create_valid_hawk_header(
+            &payload,
+            &state,
+            "GET",
+            "/storage/1.5/1/storage/tabs/asdf",
+            "localhost",
+            5000,
+        );
+        let req = TestRequest::with_state(state)
+            .header("authorization", header)
+            .method(Method::GET)
+            .uri("http://localhost:5000/storage/1.5/1/storage/tabs/asdf")
+            .param("uid", "1")
+            .param("collection", "tabs")
+            .param("bso", "asdf")
+            .finish();
+        let result = BsoRequest::extract(&req).unwrap();
+        assert_eq!(result.user_id.legacy_id, 1);
+        assert_eq!(&result.collection, "tabs");
+        assert_eq!(&result.bso, "asdf");
+    }
+
+    #[test]
+    fn test_invalid_bso_request() {
+        let payload = HawkPayload::test_default();
+        let state = make_state();
+        let header = create_valid_hawk_header(
+            &payload,
+            &state,
+            "GET",
+            "/storage/1.5/1/storage/tabs",
+            "localhost",
+            5000,
+        );
+        let req = TestRequest::with_state(state)
+            .header("authorization", header)
+            .method(Method::GET)
+            .uri("http://localhost:5000/storage/1.5/1/storage/tabs")
+            .param("uid", "1")
+            .param("collection", INVALID_COLLECTION_NAME)
```

Comment:
The name of the test is `test_invalid_bso_request`, but this suggests you're testing an invalid *collection*; should this be adding some sort of invalid bso id param as well?